src/Symfony/Component/Messenger/DispatchTask.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger;
+
+use Amp\Cache\LocalCache;
+use Amp\Cancellation;
+use Amp\Parallel\Worker\Task;
+use Amp\Sync\Channel;
+use App\Kernel;
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Dotenv\Dotenv;
+use Symfony\Component\Messenger\Exception\LogicException;
+use Symfony\Component\Messenger\Stamp\AckStamp;
+
+class DispatchTask implements Task
+{
+    private static ?LocalCache $cache = null;
+
+    public function __construct(private Envelope $envelope, private array $stamps, private readonly string $env, private readonly bool $isDebug, private readonly string $projectDir)
+    {
+    }
+
+    public function run(Channel $channel, Cancellation $cancellation): mixed
+    {
+        $container = $this->getContainer();
+        $envelope = $this->dispatch($container, $channel);
+
+        return $envelope->withoutStampsOfType(AckStamp::class);
+    }
+
+    private function dispatch(ContainerInterface $container, $channel)
+    {
+        if (!$container->has(MessageBusInterface::class)) {
+            throw new LogicException(sprintf("%s can't be found.", MessageBusInterface::class));
+        }
+
+        $messageBus = $container->get(MessageBusInterface::class);
+
+        return $messageBus->dispatch($this->envelope, $this->stamps);
+    }
+
+    private function getContainer()
+    {
+        $cache = self::$cache ??= new LocalCache();
```

Comment:
How does this cache work when you are using separate processes instead of zts/fibers? Is amphp able to maintain a cache on that static property and share it between separate PHP processes? I have a suspicion that this only works when using fibers?

Reply:
My understanding is that the same process (or thread if using `ext/parallel` will be used to handle many tasks. So a separate cache will be populated per process (and a single one will be used for threads), but it will still provide benefits.

Reply:
This is correct. `LocalCache` is thread/fiber local.